_1327/documents/utils.py

```diff
@@ -51,6 +51,14 @@ def handle_edit(request, document, formset=None, initial=None, creation_group=No
 			# save the document and also save the user and the comment the user added
 			with transaction.atomic(), revisions.create_revision():
 				document.handle_edit(cleaned_data)
+
+				# make sure to remove temp prefix of url_title
+				if document.url_title.startswith('temp_'):
+					document.url_title = document.url_title[len('temp_'):]
+				# check that there is no document already that has that url
+				if Document.objects.filter(url_title=document.url_title).exclude(id=document.id).exists():
+					document.url_title = document.generate_default_slug(document.url_title)
```

Comment:
if the `url_title` is something like `test_2` and that title already exists, this would change it to `test_2_2`, correct? maybe you could increment the existing number instead.